Week5.c++

```diff
@@ -0,0 +1,86 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+/* Link list node */
+struct node
+{
+char data;
+struct node* next;
+};
+
+// Initial parameters to this function are &head and head
+bool isPalindromeUtil(struct node **left, struct node *right)
+{
+/* stop recursion when right becomes NULL */
+if (right == NULL)
+return true;
+
+/* If sub-list is not palindrome then no need to
+check for current left and right, return false */
+bool isp = isPalindromeUtil(left, right->next);
+if (isp == false)
+return false;
+
+/* Check values at current left and right */
+bool spl = (right->data == (*left)->data);
+
+/* Move left to next node */
+*left = (*left)->next;
+
+return spl;
+}
+
+// A wrapper over isPalindromeUtil()
+bool isPalindrome(struct node *head)
+{
+isPalindromeUtil(&head, head);
+}
```

Comment:
i like your recursive solution. it's really interesting. However, I felt there are more easy to read solutions for this kind of problems. We will discuss this homework in class.